samples/bluetooth/mesh/sensor_client/src/model_handler.c

```diff
@@ -15,22 +15,29 @@
 static void sensor_cli_data_cb(struct bt_mesh_sensor_cli *cli,
 			       struct bt_mesh_msg_ctx *ctx,
 			       const struct bt_mesh_sensor_type *sensor,
-			       const struct sensor_value *value)
+			       const struct bt_mesh_sensor_value *value)
 {
 	if (sensor->id == bt_mesh_sensor_present_dev_op_temp.id) {
 		printk("Chip temperature: %s\n", bt_mesh_sensor_ch_str(value));
 	} else if (sensor->id == bt_mesh_sensor_presence_detected.id) {
-		if (value->val1) {
+		int64_t presence_detected = 0;
+
+		(void)bt_mesh_sensor_value_to_micro(value, &presence_detected);
+		if (!!presence_detected) {
 			printk("Presence detected\n");
 		} else {
 			printk("No presence detected\n");
 		}
 	} else if (sensor->id ==
 		   bt_mesh_sensor_time_since_presence_detected.id) {
-		if (!value->val1) {
-			printk("Presence detected, or under 1 second since presence detected\n");
-		} else if (value->val1 == 0xFFFF) {
+		int64_t time_since_presence_detected = 0;
+		enum bt_mesh_sensor_value_status status =
+			bt_mesh_sensor_value_to_micro(value, &time_since_presence_detected);
+
+		if (status == BT_MESH_SENSOR_VALUE_UNKNOWN) {
 			printk("Unknown last presence detected\n");
+		} else if (!time_since_presence_detected) {
+			printk("Presence detected, or under 1 second since presence detected\n");
 		} else {
 			printk("%s second(s) since last presence detected\n",
 			       bt_mesh_sensor_ch_str(value));
```

Comment:
No it prints seconds? The unit for the format is seconds

Reply:
Ahh ok, just got a little confused by the "_to_micro"

Reply:
Right, in case there still is some confusion, the `_to_micro` call is just to get a decoded integer out of it (`time_since_presence_detected`) so I can check for 0 on line 37. `value` stays as (encoded) seconds and is what is printed here.

Reply:
Makes sense!